### PR TITLE
chore: fix links in docs, treat doc warnings as errors

### PR DESCRIPTION
**Pull request by fryorcraken**

### CODE REVIEW

File: typedoc.cjs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  entryPoints: packageJson.workspaces,
+  out: "docs",
+  exclude: ["**/*.spec.ts"],
+  excludeInternal: true,
+  validation: {
+    invalidLink: true,
+    notExported: true,
+  },
+};
+packageJson.workspaces;
```
Comment (danisharora099) [inline typedoc.cjs:16]: Any specific reasons why we shifted from using `json` config to `js` config?

File: packages/core/src/lib/relay/index.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export type Observer<T extends IDecodedMessage> = {
   callback: Callback<T>;
 };
 
-export type CreateOptions = {
+export interface RelayCreateOptions extends GossipsubOpts {
```
Comment (danisharora099) [inline packages/core/src/lib/relay/index.ts:33]: are these changes under the scope of the titled PR?

File: typedoc.cjs (hunk 1 above)
Comment (fryorcraken) [reply typedoc.cjs:16]: So I could load the `entryPoints` from the `package.json`

File: packages/core/src/lib/relay/index.ts (hunk 2 above)
Comment (fryorcraken) [reply packages/core/src/lib/relay/index.ts:33]: Yes so typedoc does not handle `type`s very well especially cross repo, hence the switch to an interface.

In terms of transpiled JS code, I believe one or the other does not matter much so opted for the one typedoc is happy with.

**Review by danisharora099 — APPROVED** [review]

**Pull request merged**